torchci/pages/api/failure.ts

```diff
@@ -0,0 +1,64 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import fetchFailureSamples from "lib/fetchFailureSamples";
+import { querySimilarFailures } from "lib/drciUtils";
+import dayjs from "dayjs";
+import { RecentWorkflowsData } from "lib/types";
+import _ from "lodash";
+import { isEqual } from "lodash";
+
+interface Data {}
+
+// The maximum number of records to be returned by `more like this` failure search
+const MAX_SIZE = 1000;
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<Data>
+) {
+  const name = req.query.name as string;
+  const jobName = req.query.jobName as string;
+  const failureCaptures = JSON.parse(req.query.failureCaptures as string);
+
+  // Create a mock record to use as the input for querySimilarFailures.
+  const failure: RecentWorkflowsData = {
+    jobName: jobName,
+    name: name,
+    completed_at: dayjs().toString(),
+    failure_captures: failureCaptures,
+    failure_lines: failureCaptures,
+    // Anything goes here, it doesn't matter in this use case
+    id: "1",
+    html_url: "",
+    head_sha: "",
+  };
+
+  // The current HUD page shows the last 14 days
+  const lookbackPeriodInHours = 14 * 24;
+  const samples = await querySimilarFailures(
+    failure,
+    "",
+    lookbackPeriodInHours,
+    MAX_SIZE
+  );
+  // NB: This filter step keeps only exact matchs of the failure, this is the current
+  // behavior. However, we could consider remove this so that "slightly" different
+  // failures could be included too, like a normal search engine
+  const filteredSamples = _.filter(samples, (sample) =>
```

Comment:
If a free-form text search for failure is preferred over the current behavior, we can remove this filter step and return all relevant results

Reply:
A fuzzier match sounds would be pretty nice IMHO. Or perhaps a toggle that lets you choose between exact and fuzzy matches, which might make it easier to pinpoint when a specific error started occurring

Reply:
Yeah, a toggle sounds like an excellent idea.  I want to have exact match when I look for a specific failure, but want a fuzzy match when the error contains some unspecific parts like the Windows failure https://github.com/pytorch/pytorch/actions/runs/6593372067/attempts/1, where the file path could be different.

```
The process cannot access the file 'C:\actions-runner\_work\_actions\pytorch\pytorch\main\aten\src\ATen\native\transformers\cuda\flash_attn\kernels' because it is being used by another process
```

I'll work on the toggle.